app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,30 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @posts = Post.where(author_id: params[:user_id])
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(posts_params)
+    @post.author = current_user
+    if @post.save
+      redirect_to user_posts_path(id: @post.author_id)
+    else
+      render :new, alert: 'ERROR! something went wrong while creating the post'
+    end
```

Comment:
- [ ] When creating a new post, it's crucial to associate the post with the correct user. In this case, we fetch the user using the `user_id` parameter provided. This step ensures that the post is linked accurately with the user who is currently creating it.

However, after creating the post, there might be an issue with redirection. The code redirects to the list of posts associated with the user who initiated the creation. However, due to potential discrepancies, such as using the `current_user` method, the redirection might lead to errors. This happens when the current user and the user associated with the newly created post don't match.

To address this, we explicitly fetch the correct user using the `User.find(params[:user_id])` approach. This way, we ensure that the redirection accurately displays the list of posts created by the user who just made the new post.

- ### ERROR AFTER CREATING POST WITH ANOTHER DIFFERENT TO THE `current_user`

![image](https://github.com/Paul-tes/Blog-App/assets/87197412/ae3e317d-a1d0-4512-a3b4-7a58379a7938)


```suggestion
   def create
    @post = Post.new(posts_params)
    @post.author = User.find(params[:user_id])
    if @post.save
      redirect_to user_posts_path(id: @post.author_id)
    else
      render :new, alert: 'ERROR! something went wrong while creating the post'
    end
```